### ISSUE TITLE
Widget visibility: integrate with WooCommerce Shop Page

### ISSUE THREAD

**Issue by jesscga** [Automattic/jetpack#8343]
#### Steps to reproduce the issue
1. Add widget to widget area.
2. Under Visibility, select Show if Page is Shop

#### What I expected
Widget should appear on Shop page

#### What happened instead
Widget disappears and does not appear on any page

Workaround is using a plugin like https://wordpress.org/plugins/widget-logic/, which works as expected.

**Comment by jesscga** [comment]
Reported on 834338-zd

**Comment by jeherve** [comment]
What happens if you pick this option instead:

![screenshot 2017-12-11 at 23 06 37](https://user-images.githubusercontent.com/426388/33856315-06420356-dec8-11e7-8ff1-45c4eae863c8.png)

Does it work?

**Comment by jesscga** [comment]
Yes! Setting visibility to Show if Page is Products does make the widget appear on the Shop page.

**Comment by RonnyM82** [comment]
Why has this issue been closed? Setting it to products is far too broad reaching. The only product page I want the widget to be visible on is the Shop page... not all product archive pages.

**Comment by jeherve** [comment]
> The only product page I want the widget to be visible on is the Shop page... not all product archive pages.

The `product` custom post type created by Woocommerce should have only one archive page; the Shop page. If you have other archive pages on your site, they most likely won't be affected by this rule. If you run into issues, send us some examples and we'll take a closer look.

Thanks!

**Comment by htdat** [comment]
2079737-zen 
I think to avoid this confusion, we may silently apply the logic for `product` archive page if users select the `shop` page?

**Comment by jeherve** [comment]
>  we may silently apply the logic for product archive page if users select the shop page?

We may give this a try, however I would be afraid of creating more confusion for folks with a "Shop" page that is not actually a Product archive page today.

**Comment by xdjxdaveyx** [comment]
I found that it will not show on category or shop pages. My way around it was to **hide it** on every page that **I didn't** want the widget on!

**Comment by samiff** [comment]
I got thrown off by this. Setting the widget visibility to also show on "Taxonomy - Product categories - All pages" I could see being desirable for some.

3025302-zen

**Comment by melek** [comment]
@jeherve I opened #20693 which was a duplicate of this - thanks for tracking this down, I couldn't find this report!

Since this is an old bug (4 years old!), I don't want to mess with the tags - but I do think they should be adjusted:

* 'Need author feedback' seems a bit unlikely now, and shouldn't block fixing this issue. 
* This bug could use a priority - as I work with Woo it feels high priority, but I know I'm biased!

Want to mention that with #20057 getting worked on, it might be a convenient time to settle this one. Old bugs are all the more satisfying to squish :)

**Comment by jeherve** [comment]
> I do think they should be adjusted

You're right. I've adjusted things.

**Comment by zdenys** [comment]
Just noting that I ended up discovering the same issue https://github.com/Automattic/wp-calypso/issues/71544. 

As prevention, I think we should document the [workaround](https://github.com/Automattic/jetpack/issues/20693#issuecomment-899264467) you @jeherve mentioned above and add it both in [Jetpack.com documentation](https://jetpack.com/support/widget-visibility/) and even in [WordPress.com one](https://wordpress.com/support/widgets/#widget-visibility). 

If not, what could be a solution?

**Comment by vajrasar** [comment]
Hey all! FYI I closed [Widget Visibility Rule doesn't work for WooCommerce Shop page](https://github.com/Automattic/wp-calypso/issues/71544#top) #71544 in favor of this one.
